Add missing header comments to all pub methods in CourseLogic class #5354

rclakmal opened the pull request:
Fixes #5354 
Fixes #5355

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -115,19 +115,39 @@ public void createCourseAndInstructor(String instructorGoogleId, String courseId
     }
 
+    /**
+     * Get the {@link CourseAttributes} using the courseId.
```
Hm maybe this could be `Gets a course using courseId.`

rclakmal replied:
I skipped the 'a'. Since it returns `CourseAttributes `for the given `courseId`. Tell me if you disagree :)

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -115,19 +115,39 @@ public void createCourseAndInstructor(String instructorGoogleId, String courseId
+    /**
+     * Checks whether course is present using courseId.
+     * @param courseId
+     * @return {@code True} if course is present
```
True --> true

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -139,6 +159,12 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
+    /**
+     * Returns a list of {@link CourseDetailsBundle} for all courses a given student is enrolled in. 
```
{@link CourseDetailsBundle courseDetails}

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -139,6 +159,12 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
+    /**
+     * Returns a list of {@link CourseDetailsBundle} for all courses a given student is enrolled in. 
+     * @param googleId googleId of the student
+     * @return a list of {@link CourseDetailsBundle}
```
`{@link CourseDetailsBundle courseDetails}`

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -180,10 +206,22 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
+    /**
+     * Get a list of section names for a course using the courseId.
```
Gets

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -180,10 +206,22 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
+    /**
+     * Get a list of section names for a course using the {@link CourseAttributes}.
```
Gets

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -318,6 +362,12 @@ else if (s.team.equals(team.name)) {
     }
     
+    /**
+     * Get a list of {@link SectionDetailsBundle} for a given course using courseId.
```
Gets

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -139,6 +159,12 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
 
+    /**
+     * Returns a list of {@link CourseDetailsBundle} for all courses a given student is enrolled in. 
+     * @param googleId googleId of the student
```
@param googleId of the student

damithc replied:
**`@param googleId`** `The Google ID of the student`?

rclakmal replied:
Changed all `@**param googleIds** The Google ID of the X` format

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -115,19 +115,39 @@ public void createCourseAndInstructor(String instructorGoogleId, String courseId
+    /**
+     * Used to trigger an {@link EntityDoesNotExistException} if the course is not present. 
+     * @param courseId
+     * @throws EntityDoesNotExistException
```
Add a `@return`?

rclakmal replied:
is this necessary since it's `void` type.

JosephineKwa replied:
Oh right, my bad :p

damithc replied:
`Triggers an ....`

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -449,16 +499,28 @@ public int getTotalUnregisteredInCourse(String courseId) throws EntityDoesNotExi
-    public CourseDetailsBundle getCourseSummary(CourseAttributes cd) throws EntityDoesNotExistException {
-        Assumption.assertNotNull("Supplied parameter was null\n", cd);
+    /**
+     * Gets the {@link CourseDetailsBundle} for a course using {@link CourseAttributes}.
```
{@link CourseDetailsBundle courseDetails}

rclakmal replied:
Done. Edited all descriptions. Did not include attributeName in the `@return` tags. For an example, I think it makes more sense to say,
`@return a list of {@link CourseDetailsBundle}`
thank
`@return a list of {@link CourseDetailsBundle courseDetails}`

Tell me what you think.

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -821,6 +970,12 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
     }
     
+    /**
+     * Returns a list of courseIds for all archived courses for all instructors.
```
Returns --> Gets

rclakmal replied:
Just sharing my view, I personally think it makes much more sense to have the method summary in the description rather than in the `@return` tag to quickly grasp what method does both from understandability and structure aspects. But happy to change all, if project thinks otherwise. :)

JosephineKwa replied:
I have discussed this with Dr. Damith and he has mentioned that for simple/straightforward methods where the return value and the description could be very similar, it is not necessary to write a description, but simply write it as the return value. You could keep it as it is if you feel that it is still better to include the method summary in the description.

JosephineKwa replied:
Either way, do also change the first word of all the descriptions to either `Gets` or `Returns` where applicable for consistency :)

damithc replied:
We want to avoid duplication within the comment, between the method description and the `@return` statement. I don't mind if we keep the description and omit the `@return` or the other way around.

rclakmal replied:
@JosephineKwa @damithc Moved all the one line descriptions to return tag where ever possible to avoid duplication.

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -725,9 +850,16 @@ public void deleteCourseCascade(String courseId) {
     
-    public String getCourseStudentListAsCsv(String courseId, String googleId) throws EntityDoesNotExistException {
+    /**
+     * Returns a CSV for the details(name, email, status) of all students mapped to a given course.
```
`Returns` --> `Gets`

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -509,16 +595,35 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     }
     
+    /**
+     * Returns a list of {@link CourseAttributes} for courses a given instructor is mapped to.
```
`Returns` --> `Gets`

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -493,6 +573,12 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     
+    /**
+     * Returns a list of {@link CourseAttributes} for all courses a given student is enrolled in. 
+     * @param googleId googleId of the student
```
Edit this too

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -509,16 +595,35 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
 
+    /**
+     * Returns a list of {@link CourseAttributes} for all courses a given instructor is mapped to. 
+     * @param googleId googleId of the instructor
```
Here too

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -623,6 +728,13 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     }
     
+    /**
+     * Get a Map<CourseId, {@link CourseSummaryBundle} for all courses mapped to a given instructor.
```
Describe what the map contains, perhaps something like `Gets a map of course ids to course summaries, excluding its statistics, for all courses of a given instructor.`

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -623,6 +728,13 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
+     * Get a Map<CourseId, {@link CourseSummaryBundle} for all courses mapped to a given instructor.
+     * @param instructorId
+     * @param omitArchived if {@code True} omits all the archived courses from the return
+     * @return a Map<CourseId, {@link CourseSummaryBundle}
```
Close the `<`

rclakmal commented:
@JosephineKwa thanks for the detailed review. Addressed most of the items. I added my view for few of the changes you suggested. Will go on with them if you disagree.

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -510,16 +595,35 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     }
 
+    /**
+     * Returns a list of {@link CourseAttributes courseAttributes} for all courses a given instructor is mapped to. 
```
Instead of `is mapped to`, perhaps `belongs to`?

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -510,16 +595,35 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     }
     
+    /**
+     * Returns a list of {@link CourseAttributes courseAttributes} for courses a given instructor is mapped to.
```
Here too!

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -510,16 +595,35 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
+    /**
+     * Returns a list of {@link CourseAttributes courseAttributes} for courses a given instructor is mapped to.
+     * @param googleId The Google ID of the instructor
+     * @param omitArchived if {@code true} omits all the archived courses from the return
```
omits all the archived course from the returned list of course attributes

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -139,6 +159,12 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
+    /**
+     * Returns a list of {@link CourseDetailsBundle courseDetails} for all courses a given student is enrolled in. 
```
Sorry I made a mistake :disappointed: , this should be {@link CourseDetailsBundle course details}

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -312,14 +356,20 @@ else if (s.team.equals(team.name)) {
     }
     
+    /**
+     * Gets a list of {@link SectionDetailsBundle sectionDetails} for a given course using courseId.
```
Will need you to also edit this to be `section details` and in other descriptions too

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -494,6 +573,12 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     }
     
+    /**
+     * Returns a list of {@link CourseAttributes courseAttributes} for all courses a given student is enrolled in. 
```
If the label (i.e. `courseAttributes`) is the same as the linked value (i.e. `CourseAttributes`), perhaps you could omit the label and just leave the linked value

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -624,6 +728,13 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
+    /**
+     * Gets a Map<CourseId, {@link CourseSummaryBundle courseSummaryExcludingStatistics}> for all courses mapped to a given instructor.
+     * @param instructorId
+     * @param omitArchived if {@code true} omits all the archived courses from the return
```
from the returned map

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -471,19 +533,36 @@ public CourseDetailsBundle getCourseSummary(String courseId) throws EntityDoesNo
     
+    /**
+     * Get the {@link CourseSummaryBundle courseSummary} using the courseId.
+     * @param courseAttributes
```
This should be `courseId`

rclakmal replied:
@JosephineKwa Thanks for the review. Fixed all.

rclakmal commented:
@JosephineKwa Please re-review.

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -850,10 +837,9 @@ public void deleteCourseCascade(String courseId) {
      * @param courseId
      * @param instructorGoogleId
-     * @return a CSV String with student details
+     * @return a CSV for the details(name, email, status) of all students belongs to a given course
```
return a CSV string of the details(name, email, status) of all students belonging to a given course

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -714,9 +826,15 @@ public void deleteCourseCascade(String courseId) {
+    /**
+     * @param courseId
+     * @param instructorGoogleId
+     * @return a CSV for the details(name, email, status) of all students belongs to a given course
```
return a CSV string of the details(name, email, status) of all students belonging to a given course

JosephineKwa commented:
Just a minor change suggested

rclakmal commented:
@JosephineKwa Done.

damithc commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -254,7 +289,13 @@ public SectionDetailsBundle getSectionForCourse(String section, String courseId)
     
-    public List<SectionDetailsBundle> getSectionsForCourse(CourseAttributes course, CourseDetailsBundle cdd) {
+    /**
+     * Returns a list of {@link SectionDetailsBundle section details} for a given course using course attributes and course details bundle.
```
This line seems a bit too long. I have to scroll horizontal to read it in GitHub diff view.

damithc commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -463,19 +519,35 @@ public CourseDetailsBundle getCourseSummary(String courseId) throws EntityDoesNo
     }
     
+    /**
+     * Returns the {@link CourseSummaryBundle course summary}, including its feedback sessions using the given {@link InstructorAttributes}.
```
Line too long?

damithc commented:
Cosmetic changes requested:
1. The line length seems to be slightly too long in a few places. I didn't mark all of them though.
2. I noticed some duplication between method description and the @return statement in a few places.

rclakmal commented:
@damithc addressed both. Please re review.

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -463,19 +519,35 @@ public CourseDetailsBundle getCourseSummary(String courseId) throws EntityDoesNo
-            InstructorAttributes instructor) throws EntityDoesNotExistException {
-        CourseSummaryBundle courseSummary = getCourseSummaryWithoutStats(instructor.courseId);
-        courseSummary.feedbackSessions.addAll(feedbackSessionsLogic.getFeedbackSessionListForInstructor(instructor));
+            InstructorAttributes instructorAttributes) throws EntityDoesNotExistException {
```
I missed this out earlier, but I feel that the changes should be related to this PR, i.e. adding the javadoc comments. Perhaps if you'd like to refactor/rename variables, create a separate issue for it and give your suggestions on what the variables should be renamed to in that issue?

JosephineKwa replied:
In most of the files, EntityAttributes variables are simply named as the entity (without "Attributes"),  so changing it here would make it somewhat inconsistent with the rest of the files

damithc replied:
I think we used the shorter form to reduce line length.  When the lines get too long, they need wrapping, and the code gets messier.
Yes, let's keep it consistent with the rest.

rclakmal replied:
@damithc @JosephineKwa Reverted all attribute name changes. Ready for review.

JosephineKwa commented:
@damithc would also like to seek your opinion on the renamed variables

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -182,20 +207,30 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     /**
-     * Get list of section names for a course with or without a need to check if the course is existent
      * @param courseId Course ID of the course
      * @param hasCheckIsPresent Determine whether it is necessary to check if the course exists
```
This should be `isCourseVerified`

JosephineKwa replied:
Though this wasn't due to your changes, could you also help fix this by changing this to `isCourseVerified` as well?

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -463,19 +519,35 @@ public CourseDetailsBundle getCourseSummary(String courseId) throws EntityDoesNo
     }
 
+    /**
+     * @param courseAttributes
```
Change this to `course` too :)

JosephineKwa commented on src/main/java/teammates/logic/core/CoursesLogic.java:
```diff
@@ -502,15 +579,31 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
     
+    /**
+     * @param googleId The Google ID of the instructor
+     * @param omitArchived if {@code true},, omits all the archived courses from the return
```
There are 2 commas here

JosephineKwa commented:
Should be good to merge after the comments are addressed :+1:

rclakmal commented:
@JosephineKwa  Done :) Thanks for the review

wkurniawan07 commented:
@rclakmal can you update this one and do any necessary fixes, so it can be merged?

rclakmal commented:
@wkurniawan07 Done.

wkurniawan07 commented:
@rclakmal perfect, I'll merge it once Travis gives a green.
